kasa/tests/fakeprotocol_smart.py

```diff
@@ -249,6 +249,17 @@ def _edit_preset_rules(self, info, params):
         info["get_preset_rules"]["states"][params["index"]] = params["state"]
         return {"error_code": 0}
 
+    def _set_temperature_unit(self, info, params):
+        """Set or remove values as per the device behaviour."""
+        unit = params["temp_unit"]
+        if unit not in {"celsius", "fahrenheit"}:
+            raise ValueError(f"Invalid value for temperature unit {unit}")
+        if "temp_unit" not in info["get_device_info"]:
+            return {"error_code": SmartErrorCode.UNKNOWN_METHOD_ERROR}
+        else:
+            info["get_device_info"]["temp_unit"] = unit
+            return {"error_code": 0}
```

Comment:
Changing the temperature unit should be done in a separate PR, the question is how to encode this? Have an enum? This is also related to the one missing piece to support temperature sensors (i.e., we don't currently expose the unit properly to downstreams).

Reply:
Is this still something we want to do in a separate PR? It's just the test framework iirc.